src/view3d.cpp

```diff
@@ -185,61 +209,155 @@ void View3D::centerOnGeometry(const QModelIndex& index)
+
+    m_meshEdgeShader.reset(new ShaderProgram());
     m_meshEdgeShader->setShaderFromSourceFile("shaders:meshedge.glsl");
-    m_incrementalFramebuffer = allocIncrementalFramebuffer(width(), height());
+
+#ifdef DISPLAZ_USE_QT4
+    double dPR = 1.0;
+#else
+    double dPR = devicePixelRatio();
+#endif
+    int w = width() * dPR;
+    int h = height() * dPR;
+
+    m_incrementalFramebuffer = allocIncrementalFramebuffer(w, h);
+
+    glFrameBufferStatus(m_incrementalFramebuffer);
+    glCheckError();
+
     const GeometryCollection::GeometryVec& geoms = m_geometries->get();
     for (size_t i = 0; i < geoms.size(); ++i)
-        geoms[i]->initializeGL();
+    {
+        if (m_boundingBoxShader->isValid())
+        {
+            // TODO: build a shader manager for this
+            geoms[i]->setShaderId("boundingbox", m_boundingBoxShader->shaderProgram().programId());
+            geoms[i]->setShaderId("meshface", m_meshFaceShader->shaderProgram().programId());
+            geoms[i]->setShaderId("meshedge", m_meshEdgeShader->shaderProgram().programId());
+            geoms[i]->initializeGL();
+        }
+    }
+
+    PointViewerMainWindow * pv_parent = dynamic_cast<PointViewerMainWindow *>(parentWidget());
+    pv_parent->openShaderFile("shaders:las_points.glsl");
+
+    setFocus();
+
+    resizeGL(w,h);
+    updateGL();
 }
 
 
 void View3D::resizeGL(int w, int h)
 {
+    //Note: This function receives "device pixel sizes" for correct OpenGL viewport setup
+    //      Under OS X with retina display, this becomes important as it is 2x the width() or height()
+    //      of the normal window (there is a devicePixelRatio() function to deal with this).
+
     if (m_badOpenGL)
         return;
+
     // Draw on full window
     glViewport(0, 0, w, h);
-    m_camera.setViewport(QRect(0,0,w,h));
+
+#ifdef DISPLAZ_USE_QT4
+    double dPR = 1.0;
+#else
+    double dPR = devicePixelRatio();
+#endif
+    m_camera.setViewport(QRect(0,0,double(w)/dPR,double(h)/dPR));
+
     m_incrementalFramebuffer = allocIncrementalFramebuffer(w,h);
+
+    glCheckError();
+
+    // without this explicit call, we get very strange artifacts during resizing
+    //updateGL();
 }
 
 
-std::unique_ptr<QGLFramebufferObject> View3D::allocIncrementalFramebuffer(int w, int h) const
+unsigned int View3D::allocIncrementalFramebuffer(int w, int h) const
 {
+    if (w < 1 || h < 1)
+        return 0;
+
+    //should we really delete this every time?
+    GLuint fb;
+    glDeleteFramebuffers(1, &fb);
+
+    const QGLFormat fmt = context()->format();
+
+    glGenFramebuffers(1, &fb);
+    glBindFramebuffer(GL_DRAW_FRAMEBUFFER, fb);
+
     // TODO:
     // * Should we use multisampling 1 to avoid binding to a texture?
-    const QGLFormat fmt = context()->format();
-    QGLFramebufferObjectFormat fboFmt;
-    fboFmt.setAttachment(QGLFramebufferObject::Depth);
+    // * but why would you want to avoid binding to a texture? It should be more efficient than copying the fb
```

Comment:
Let's have the conversation about this in the pull request, rather than the code comments :-)